src/PeopleLookup.Mvc/Controllers/HomeController.cs

```diff
@@ -60,14 +60,26 @@ public async Task<ActionResult> Index(BulkModel model)
                 matches = System.Text.RegularExpressions.Regex.Matches(model.BulkEmail, regexEmailPattern,
                         System.Text.RegularExpressions.RegexOptions.IgnoreCase);
 
+                var results1 = new List<Task<SearchResult>>();
                 foreach (var match in matches)
                 {
-                    var result = await _identityService.Lookup(match.ToString());
+                    //var result = await _identityService.Lookup(match.ToString());
+                    //if (!allowSensitiveInfo)
+                    //{
+                    //    result.HideSensitiveFields();
+                    //}
+                    //model.Results.Add(result);
+                    results1.Add(_identityService.Lookup(match.ToString()));
+                }

```

Comment:
You can replace this whole thing with `matches.Select(m => _identityService.Lookup(m.ToString()));` . No need for the temp var or for loop, just project into an array of Tasks.